### PR TITLE
Legacy errors

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by sergiusens** [comment]
This should solve https://bugs.launchpad.net/snapcraft/+bug/1836182

File: snapcraft/internal/build_providers/_multipass/_multipass.py [hunk 1]
```diff
@@ -189,6 +191,23 @@ def retrieve_snap(self) -> str:
         self._multipass_cmd.copy_files(source=source, destination=self.snap_filename)
         return self.snap_filename
 
+    def pull_file(self, name: str, destination: str, delete: bool = False) -> None:
+        # TODO add instance check.
```
Comment (cjp256) [inline snapcraft/internal/build_providers/_multipass/_multipass.py]: instance, as in if it's running?

**Comment by cjp256** [comment]
Those errors look familiar, I think you also need to pick: 
```
commit 290ec0ea7e60ac5dd8c4dfb6970aa73a9e3f0d91
Author: Chris Patterson <chris.patterson@canonical.com>
Date:   Thu Jun 27 18:25:16 2019 -0400

    echo: respect SNAPCRAFT_HAS_TTY for is_tty_connected()
    
    Enable environment fixture SNAPCRAFT_HAS_TTY=y for integration tests,
    to ensure pexpect can interact with prompt(), etc.
    
    Signed-off-by: Chris Patterson <chris.patterson@canonical.com>
```

**Comment by sergiusens** [comment]
When using the snap, before:
```
Pulling foo
Sorry, an error occurred in Snapcraft.
We would appreciate it if you anonymously reported this issue.
No other data than the traceback ('/tmp/tmpakpc3xhn/trace.txt') and the version of snapcraft in use will be sent.
Would you like to send this error data? (Yes/No/Always) [no]:
```
and after:
```
Pulling foo
Sorry, an error occurred in Snapcraft:
404 Client Error: Not Found for url: https://sergiusens.org/foo.tar
We would appreciate it if you anonymously reported this issue.
No other data than the traceback and the version of snapcraft in use will be sent.
Would you like to send this error data? (Yes/No/Always/View) [no]:
```

**Pull request merged**